wordpress-mobile/WordPress-iOS
# PostViewController: Adds quick Publish / Draft item to options menu.

Fixes #6121. 

Adds Publish Now / Submit for Review / Save as Draft options to the options menu in the editor.

* If the post is new and the status is currently set to Publish, then the options menu will contain a "Save as Draft" option. This toggles the post status and updates the content of the local post.
* If the post's status is currently set to Draft and the user has permission to publish posts, the options 
menu will contain a "Publish Now" option.
* If the post's status is currently set to Draft and the user does not have permission to publish posts, the options menu will contain a "Submit for Review" option.

To test:

* Try the above scenarios out in the editor, and others besides to ensure it's not appearing when it shouldn't (and is appearing when it should)!
* Ensure that the Post / Update / Save button in the navigation bar updates accordingly when saving as draft.

Needs review: @kurzee 

@rachelmcr @mattmiklic Would you two also care to take a look?
